EIPS/eip-2929.md

```diff
@@ -0,0 +1,165 @@
+
+#### Goerli analysis
+
+A more precise analysis can be found by scanning Goerli transactions, as done by Martin Swende here: https://github.com/holiman/gasreprice
+
+The conclusion is that on average gas costs increase by ~2.36%. One major contributing factor to reducing gas costs is that a large number of contracts inefficiently read the same storage slot multiple times, which leads to this EIP giving a few transactions gas cost _savings_ of over 10%.
+
+## Backwards Compatibility
+
+These gas cost increases may potentially break contracts that depend on fixed gas costs; see the security considerations section for details and arguments for why we expect the total risks to be low and how if desired they can be reduced further.
+
+## Test Cases
+
+Some test cases can be found here: https://gist.github.com/holiman/174548cad102096858583c6fbbb0649a
+
+Ideally we would test the following:
+
+* SLOAD the same storage slot {1, 2, 3} times
+* CALL the same address {1, 2, 3} times
+* (SLOAD | CALL) in a sub-call, then revert, then (SLOAD | CALL) the same (storage slot | address) again
+* Sub-call, SLOAD, sub-call again, revert the inner sub-call, SLOAD the same storage slot
+* SSTORE the same storage slot {1, 2, 3} times, using all combinations of zero/nonzero for original value and the value being set
+* SSTORE then SLOAD the same storage slot
+* `OP_1` then `OP_2` to the same address where `OP_1` and `OP_2` are all combinations of (\*CALL, EXT\*, SELFDESTRUCT) 
+* Try to CALL an address but with all possible failure modes (not enough gas, not enough ETH...), then (CALL | EXT*) that address again successfully
+
+## Implementation
+
+A WIP early-draft implementation for Geth can be found here: https://github.com/holiman/go-ethereum/tree/access_lists
+
+## Security Considerations
+
+As with any gas cost increasing EIP, there are three possible cases where it could cause applications to break:
+
+1. Fixed gas limits to sub-calls in contracts
+2. Applications relying on contract calls that consume close to the full gas limit
+3. The 2300 base limit given to the callee by ETH-transferring calls
+
+These risks have been studied before in the context of an earlier gas cost increase, EIP 1884. See [Martin Swende's earlier report](https://github.com/holiman/eip-1884-security) and [Hubert Ritzdorf's analysis](https://gist.github.com/ritzdorf/1c6bd72955391e831f8a397d3152b4e0/) focusing on (1) and (3). (2) has received less analysis, though one can argue that it is very unlikely both because applications tend to very rarely use close to the entire gas limit in a transaction, and because gas limits were very recently raised from 10 million to 12.5 million. EIP 1884 in practice [did lead to a small number of contracts breaking](https://www.coindesk.com/ethereums-istanbul-upgrade-will-break-680-smart-contracts-on-aragon) for this reason.
```

Comment:
```suggestion
These risks have been studied before in the context of an earlier gas cost increase, EIP-1884. See [Martin Swende's earlier report](https://github.com/holiman/eip-1884-security) and [Hubert Ritzdorf's analysis](https://gist.github.com/ritzdorf/1c6bd72955391e831f8a397d3152b4e0/) focusing on (1) and (3). (2) has received less analysis, though one can argue that it is very unlikely both because applications tend to very rarely use close to the entire gas limit in a transaction, and because gas limits were very recently raised from 10 million to 12.5 million. EIP-1884 in practice [did lead to a small number of contracts breaking](https://www.coindesk.com/ethereums-istanbul-upgrade-will-break-680-smart-contracts-on-aragon) for this reason.
```

Reply:
Per the EIP-1 style guide, EIPs should be cited as `EIP-N` where `N` is the EIP number.